File: leetcode/c#/PalindromeNumber.cs
using System;

public class PalindromeNumber {
  public struct Test {
    public int input;
    public bool output;
    public Test(int input, bool output) {
      this.input = input;
      this.output = output;
    }
  }

  public static Test[] tests = new Test[]{
    new Test(5, true),
    new Test(121, true),
    new Test(-121, false),
    new Test(10, false),
  };

  public bool IsPalindrome(int x) {
    if (x < 0) return false;
    var s = x.ToString();
    var n = s.Length;
    for (int i = 0; i < n/2; i++) {
      if (s[i] != s[n - i - 1]) return false;
    }
    return true;
  }

  public static void Main(string[] args) {
    Console.WriteLine("Running...");
    var m = new PalindromeNumber();
    foreach (var t in tests) {
      var actual = m.IsPalindrome(t.input);
      if (actual == t.output) {
        Console.WriteLine("Success");
      } else {
        Console.WriteLine(String.Format("Failure: {0} != {1}", actual, t.output));
      }
    }
  }
}

File: leetcode/go/PalindromeNumber.go
package main

import (
	"fmt"
	"math"
	"strconv"
)

func isPalindrome2(x int) bool {
	if x < 0 {
		return false
	} else if x == 0 {
		return true
	}
	n := int(math.Log10(float64(x))) + 1
	v := make([]int, n)
	for i := 0; i < n; i++ {
		v[i] = x % 10
		x /= 10
	}
	for i := 0; i < len(v)/2; i++ {
		if v[i] != v[len(v)-i-1] {
			return false
		}
	}
	return true
}

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	s := strconv.FormatInt(int64(x), 10)
	n := len(s)
	for i := 0; i < n/2; i++ {
		if s[i] != s[n-i-1] {
			return false
		}
	}
	return true
}

type test struct {
	in  int
	out bool
}

var tests = []test{
	{5, true},
	{121, true},
	{-121, false},
	{10, false},
}

func main() {
	fmt.Println("Running...")
	for _, t := range tests {
		actual := isPalindrome(t.in)
		if actual == t.out {
			fmt.Println("Success")
		} else {
			fmt.Printf("Failure: %v != %v\n", actual, t.out)
		}
	}
}
